### DISCUSSION TITLE
Allow deployment in a single role

### DISCUSSION THREAD

**Discussion by wenderjean** [basecamp/kamal#698] [Ideas]
Hi guys!

So, by the time I'm writing this suggestion I'm trying to test a specific configuration for Sidekiq and to do that I need to deploy a new instance, no surprise up here :) but... It seems like Kamal does not allow us to run a commands like these ones:

`kamal deploy -P -d production --version=xyz --roles=worker`
`kamal redeploy -d production --version=xyz --roles=worker`

Initially I assumed it was a bug considering when the command executes, the error I receive is that Kamal cannot find `.kamal/env/roles/app-web-production.env` but after reading this DHH's comment [here](https://github.com/basecamp/kamal/issues/191#issuecomment-1501728562) I'm not that sure if this should even be possible assuming it could open up room for non-web applications deployment.

What do you guys think about this feature? The scenario itself seems doable.

**Comment by morgoth** [comment]
I got the same error, vide https://github.com/basecamp/kamal/issues/709
